src/lap_stats/lap_stats.c

```diff
@@ -297,86 +313,99 @@ static void sectorBoundaryEvent(const GpsSnapshot *gpsSnapshot)
     g_lastSectorTime = millis - g_lastSectorTimestamp;
     g_lastSectorTimestamp = millis;
     g_lastSector = g_sector;
-    ++g_sector;
+    g_at_sector = true;
+    update_sector_geo_circle(++g_sector);
 }
 
 /**
  * All logic associated with determining if we are at the finish line.
  */
-static void processFinishLogic(const GpsSnapshot *gpsSnapshot,
-                               const Track *track, const float targetRadius)
+static void process_finish_logic(const GpsSnapshot *gpsSnapshot)
 {
     if (!lapstats_lap_in_progress())
         return;
+
     if (!isGeoTriggerTripped(&g_finish_geo_trigger))
         return;
 
     const GeoPoint point = gpsSnapshot->sample.point;
-    const GeoPoint finishPoint = getFinishPoint(track);
-    const struct GeoCircle finishCircle = gc_createGeoCircle(finishPoint,
-                                          targetRadius);
-    if (!gc_isPointInGeoCircle(&point, finishCircle))
+    if (!gc_isPointInGeoCircle(&point, g_geo_circles.finish))
         return;
 
     // If we get here, then we have completed a lap.
     lap_finished_event(gpsSnapshot);
 }
 
+static void process_start_normal(const GpsSnapshot *gpsSnapshot)
+{
+        const GeoPoint point = gpsSnapshot->sample.point;
+        if (!gc_isPointInGeoCircle(&point, g_geo_circles.start))
+                return;
+
+        const tiny_millis_t time = gpsSnapshot->deltaFirstFix;
+        lap_started_event(time, &point, 0);
+}
+
+static void process_start_launch_control(const GpsSnapshot *gpsSnapshot)
+{
+        lc_supplyGpsSnapshot(gpsSnapshot);
+        if (!lc_hasLaunched())
+                return;
+
+        /*
+         * If here, then launch control has determined that the lap
+         * has begun.  Launch control unfortunately causes a delay in
+         * reporting since it has to detect a launch before it can
+         * report it.  Hence we must retroactively calculate the
+         * distance we have traveled and the time elapsed since we
+         * launched from start as it is before this point in time.
+         */
```

Comment:
So in other launch control designs, we would have a circular buffer that would contain all of the channel samples so that when we've detected we've launched, we can include all of that previous data back to the point where we actually launched.  We don't use a big circular buffer right now, so we must be using an alternate approach.

Here, it looks like you're just calculating the distance from where we launched from - and the "lap" starts with a pre-calculated distance value?